src/components/e-table.vue

```diff
@@ -95,64 +273,160 @@
+        type: Array as PropType<IETableColumn[]>,
         required: true,
       },
+
+      /**
+       * Accepts a method to generate a link for each row (except for columns with 'onClick' callback).
+       */
+      rowLink: {
+        type: Object as PropType<IRowLink>,
+        default: null,
+      },
+
+      /**
+       * Adds a checkbox to each row to allow selections.
+       */
+      selectable: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Disables the tables interaction features.
+       */
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Determines if the table should show an additional column to toggle row details.
+       */
+      hasDetailRows: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Allows to change the identifier property for a row item by which it will be identified internally.
+       */
+      itemIdentifier: {
+        type: String,
+        default: 'id',
+      },
+    },
+    emits: {
+      'update:selected': (payload: unknown): boolean => Array.isArray(payload),
+    },
+
+    setup(): ISetup {
+      const toggleButton = ref();
+
+      return {
+        ...useUuid(),
+        toggleButton,
+      };
     },
     data(): IData {
       return {
-        /**
-         * the currently selected 'column' to be sorted.
-         */
+        showSortingOptions: true,
         sortBy: null,
-
-        /**
-         * Holds to sort direction in case a 'sortBy' is active.
-         */
         sortAscending: true,
+        enableRowLinks: false,
+        hasSelection: false,
+        toggleButtonHeight: 0,
+        expandedRows: [],
       };
     },
 
     computed: {
+      /**
+       * Model Value for toggle all checkbox.
+       */
+      toggleAllValue: {
+        get(): boolean {
+          return !!this.selectedInternal.length;
+        },
+        set(): void {
+          if (this.selectedInternal.length) {
+            this.selectedInternal = [];
+          } else {
+            this.selectedInternal = this.items.map(item => item[this.itemIdentifier]);
+          }
+        },
+      },
+
+      /**
+       * Checks if current viewport is mobile (<= sm).
+       */
+      isMobile(): boolean {
+        return !this.$viewport.isSm;
+      },
+
+      /**
+       * Manages changes for the 'select' prop.
+       */
+      selectedInternal: {
+        get(): unknown[] {
+          return this.selected.map((item: IETableItem): unknown => item[this.itemIdentifier]);
+        },
+        set(itemIds: unknown[]): void {
+          this.$emit('update:selected', this.items.filter(item => itemIds.includes(item[this.itemIdentifier])));
+        },
+      },
+
+      /**
+       * Handles changes to the 'expandedRows' prop.
+       */
+      expandedRowsComputed: {
+        get(): unknown[] {
+          return this.expandedRows.map(item => item[this.itemIdentifier]);
```

Comment:
todo: return value is an array of `string | number`. Here you did not define the type of `item`. How should we handle such cases? I think it should not required because the type should be given by `expandedRows`.